lib/Network/Wai/Middleware/Shibboleth.hs

```diff
@@ -0,0 +1,69 @@
+--
+-- Copyright © 2013-2015 Anchor Systems, Pty Ltd and Others
+--
+-- The code in this file, and the program it is a part of, is
+-- made available to you by its authors as open source software:
+-- you can redistribute it and/or modify it under the terms of
+-- the 3-clause BSD licence.
+--
+
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE RecordWildCards   #-}
+
+-- | Description: WAI middleware for Shibboleth-protected applications.
+module Network.Wai.Middleware.Shibboleth where
```

Comment:
I wonder if shibboleth is too specific here, perhaps just "ProxiedHeader" or something